### PR TITLE
Properly build-tag the Linux kube-proxy backend code

### PR DESCRIPTION
**Pull request by danwinship**
#### What type of PR is this?
/kind cleanup
/sig network
/priority backlog

#### What this PR does / why we need it:
With #121722 merged we can now properly tag the linux kube-proxy backends as being for "`linux`" rather than "`!windows`".

(@aroradaman this does not also fix `GetKernelVersion` because the kubelet code doesn't currently have its caller split out into a linux-only file, but if you wanted to fix that now, you could.)

#### Which issue(s) this PR fixes:
None, but see discussion in https://github.com/kubernetes/kubernetes/pull/120995#discussion_r1374955990

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: pkg/proxy/ipvs/safe_ipset.go [hunk 1]
```diff
@@ -1,104 +0,0 @@
...
-
-package ipvs
-
-import (
-	"sync"
-
-	"k8s.io/kubernetes/pkg/proxy/ipvs/ipset"
-)
-
-type safeIpset struct {
```
Comment (aojea) [inline pkg/proxy/ipvs/safe_ipset.go:25]: the problem is that there can be two different execs in parallel https://github.com/kubernetes/kubernetes/pull/80627/files#r318400672

File: pkg/proxy/ipvs/proxier.go [hunk 2]
```diff
@@ -489,10 +489,8 @@ func NewDualStackProxier(
 	initOnly bool,
 ) (proxy.Provider, error) {
 
-	safeIpset := newSafeIpset(ipset)
```
Comment (aojea) [inline pkg/proxy/ipvs/proxier.go:492]: oh, so they share the same exec 
https://github.com/kubernetes/kubernetes/pull/122373/commits/b69510b0690073510fdb787f34687c4d0a40e358#r1441864562

**Comment by liggitt** [comment]
this breaks build on darwin... how did this pass typecheck?

```
+ make -C /Users/liggitt/go/src/k8s.io/kubernetes 'WHAT=cmd/kubectl cmd/kube-apiserver cmd/kube-controller-manager cmd/cloud-controller-manager cmd/kubelet cmd/kube-proxy cmd/kube-scheduler'
+++ [0118 08:30:36] Building go targets for darwin/arm64
    k8s.io/kubernetes/cmd/kubectl (non-static)
    k8s.io/kubernetes/cmd/kube-apiserver (static)
    k8s.io/kubernetes/cmd/kube-controller-manager (static)
    k8s.io/kubernetes/cmd/cloud-controller-manager (non-static)
    k8s.io/kubernetes/cmd/kubelet (non-static)
    k8s.io/kubernetes/cmd/kube-proxy (static)
    k8s.io/kubernetes/cmd/kube-scheduler (static)
# k8s.io/kubernetes/cmd/kube-proxy/app
cmd/kube-proxy/app/server.go:145:4: o.addOSFlags undefined (type *Options has no field or method addOSFlags)
cmd/kube-proxy/app/server.go:281:4: o.platformApplyDefaults undefined (type *Options has no field or method platformApplyDefaults)
cmd/kube-proxy/app/server.go:381:9: undefined: platformCleanup
cmd/kube-proxy/app/server.go:538:14: undefined: initForOS
cmd/kube-proxy/app/server.go:649:10: s.platformSetup undefined (type *ProxyServer has no field or method platformSetup)
cmd/kube-proxy/app/server.go:654:61: s.platformCheckSupported undefined (type *ProxyServer has no field or method platformCheckSupported)
cmd/kube-proxy/app/server.go:673:21: s.createProxier undefined (type *ProxyServer has no field or method createProxier)

```

**Comment by liggitt** [comment]
oh, https://github.com/kubernetes/kubernetes/pull/121722 broke typecheck on darwin :)

**Comment by liggitt** [comment]
fixed local-up-cluster on darwin in https://github.com/kubernetes/kubernetes/pull/122843 ... I'm still not super happy that we have packages that fail to build on some platforms

**Comment by danwinship** [comment]
> I'm still not super happy that we have packages that fail to build on some platforms

Why exactly? Kube-proxy used to _build_ but it never _worked_. (Even the old userspace proxy depended heavily on iptables.)

We _could_ add a new `server_others.go` that does

```
func (s *ProxyServer) platformSetup() error {
        return fmt.Errorf("kube-proxy is not supported on this platform")
}
```

But who would that actually help?

**Comment by liggitt** [comment]
> But who would that actually help?

people running `make` on darwin, which explodes right now

```
$ make
+++ [0119 07:42:09] Building go targets for darwin/arm64
    k8s.io/kubernetes/cmd/kube-proxy (static)
    k8s.io/kubernetes/cmd/kube-apiserver (static)
    k8s.io/kubernetes/cmd/kube-controller-manager (static)
    k8s.io/kubernetes/cmd/kubelet (non-static)
    k8s.io/kubernetes/cmd/kubeadm (static)
    k8s.io/kubernetes/cmd/kube-scheduler (static)
    k8s.io/component-base/logs/kube-log-runner (static)
    k8s.io/kube-aggregator (static)
    k8s.io/apiextensions-apiserver (static)
    k8s.io/kubernetes/cluster/gce/gci/mounter (static)
    k8s.io/kubernetes/cmd/kubectl (non-static)
    k8s.io/kubernetes/cmd/kubectl-convert (static)
    github.com/onsi/ginkgo/v2/ginkgo (non-static)
    k8s.io/kubernetes/test/e2e/e2e.test (test)
    k8s.io/kubernetes/test/conformance/image/go-runner (non-static)
    k8s.io/kubernetes/cmd/kubemark (static)
    github.com/onsi/ginkgo/v2/ginkgo (non-static)
# k8s.io/kubernetes/cmd/kube-proxy/app
cmd/kube-proxy/app/server.go:145:4: o.addOSFlags undefined (type *Options has no field or method addOSFlags)
cmd/kube-proxy/app/server.go:281:4: o.platformApplyDefaults undefined (type *Options has no field or method platformApplyDefaults)
cmd/kube-proxy/app/server.go:381:9: undefined: platformCleanup
cmd/kube-proxy/app/server.go:538:14: undefined: initForOS
cmd/kube-proxy/app/server.go:649:10: s.platformSetup undefined (type *ProxyServer has no field or method platformSetup)
cmd/kube-proxy/app/server.go:654:61: s.platformCheckSupported undefined (type *ProxyServer has no field or method platformCheckSupported)
cmd/kube-proxy/app/server.go:673:21: s.createProxier undefined (type *ProxyServer has no field or method createProxier)
```

I think it's much better to make our top-level commands build consistently and indicate at runtime if specific platforms aren't supported

**Comment by liggitt** [comment]
Proposed a revert in https://github.com/kubernetes/kubernetes/pull/122878

**Pull request merged**